server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -0,0 +1,251 @@
+import previousWorkExperienceTypeScreenOrderComparator from '../previousWorkExperienceTypeScreenOrderComparator'
+
+export default class PreviousWorkExperienceTypesUpdateController extends PreviousWorkExperienceTypesController {
+  constructor(private readonly inductionService: InductionService) {
+    super()
+  }
+
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/work-and-interests`
+  }
+
+  getBackLinkAriaText(_req: Request): string {
+    return 'Back to <TODO - check what CIAG UI does here>'
+  }
+
+  submitPreviousWorkExperienceTypesForm: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary, inductionDto } = req.session
+
+    req.session.previousWorkExperienceTypesForm = { ...req.body }
+    if (!req.session.previousWorkExperienceTypesForm.typeOfWorkExperience) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = []
+    }
+    if (!Array.isArray(req.session.previousWorkExperienceTypesForm.typeOfWorkExperience)) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = [
+        req.session.previousWorkExperienceTypesForm.typeOfWorkExperience,
+      ]
+    }
+    const { previousWorkExperienceTypesForm } = req.session
+
+    const errors = validatePreviousWorkExperienceTypesForm(previousWorkExperienceTypesForm, prisonerSummary)
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      return res.redirect(`/prisoners/${prisonNumber}/induction/previous-work-experience`)
+    }
+
+    if (
+      noChangesToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm) ||
+      onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)
+    ) {
+      if (onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)) {
+        logger.debug(
+          'Previous Work Experiences changes are only removed types so Induction can be updated without starting a flow',
+        )
+        // call the service to update the Induction
+        try {
+          await this.updateInduction(inductionDto, previousWorkExperienceTypesForm, req, prisonerSummary)
+        } catch (e) {
+          return next(createError(500, `Error updating Induction for prisoner ${prisonNumber}. Error: ${e}`))
+        }
+      } else {
+        logger.debug('No changes to Previous Work Experiences were submitted')
+      }
+
+      req.session.previousWorkExperienceTypesForm = undefined
+      req.session.inductionDto = undefined
+      return res.redirect(`/plan/${prisonNumber}/view/work-and-interests`)
+    }
+
+    /* We need to show the Details page for each of:
+         - previous work experience on the original induction that has not been removed by the form submission
```

Comment:
Deciding when to show the details page is probably up for debate, but I guess you don't want to ask for guidance from our illustrious design/UR team?!

I'm conscious I suggested we ask for the details of each job type (new and existing) when they make changes (i.e. whey they don't just remove some). I think is the right decision, but there is an inherent inconsistency here...

What happens if they want to remove one and edit the job role of another? I assume they'd have to remove one from the list and then go back to the "Work and interests" page, before clicking on the change link for "Specific job role of interest". That might be fine, but if they added a job type, then they'd also be able to edit the existing job roles in the same flow. Does that make sense?

Reply:
> What happens if they want to remove one and edit the job role of another? 
They would do the "edit" on the Change link of the job type in question, and not via this change link (which is for the list of job types)

I think the complexity is actually because you talked me into changing the process slightly in this respect, where the "queue" of detail pages is to include all job types on the induction, even those that you've not changed here

Might be easier to chat through on a call .. 🤔